### PR TITLE
RF=2 metrics bug fix

### PR DESCRIPTION
**Pull request by jcacciatore**
When the server topology has replication factor of 2 AND there are dyno client instances that are located in a rack that has no corresponding dynomite hosts, the failover metrics are not accurate since every request is viewed as failing over to a remote rack.

For example, if dynomite hosts are in us-east-1c and us-east-1d but dyno client instances are in 1c, 1d, **and also 1e** all requests from dyno client instances in 1e would be recorded as failover requests.

This PR fixes this issue.

### CODE REVIEW

File: dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java [hunk 1]
```diff
@@ -326,6 +326,37 @@ public void testGetConnectionsFromRingWhenHostDown() throws Exception {
 		verifyExactly(hostnames, "h5", "h6");
 	}
 
+    @Test
+    public void testReplicationFactorOf3WithDupes() {
```
Comment (ipapapa) [inline dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java]: What does Dupes mean? Do we need a Test for RF2?

**Comment by ipapapa** [comment]
Based on my understanding of the implementation (which may be totally off), if there is no available node in the local rack, we assume that we are using RF=2. This may create some corner cases, so I was wondering whether we can cross-validate whether the Dynomite server is RF=2 through cluster_describe.

File: dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java (hunk 1 above)
Comment (jcacciatore) [inline dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java]: There is a test for RF2, [here](https://github.com/jcacciatore/dyno/blob/3b482d9d0ed41a4ccd6bb595dc79edc779686d21/dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java#L382).

The client contacts every dynomite server in determining the topology therefore it sees a lot of duplicate info.

**Comment by jcacciatore** [comment]
@ipapapa The logic to determine the replication factor is based on the cluster_describe from the side-car (currently). Where do you see an assumption of RF2 ?

File: dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java (hunk 1 above)
Comment (ipapapa) [inline dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java]: ha, I See it in the "new Host". Thanks covered.
